main/modes/games/cosCrunch/cosCrunch.c

```diff
@@ -325,6 +329,8 @@ static void cosCrunchExitMode(void)
     unloadMidiFile(&cc->gameBgm);
     unloadMidiFile(&cc->gameOverBgm);
 
+    freeHighScoreSonas(&cc->highScores, cc->highScoreSonas);
```

Comment:
It's good to free here in `cosCrunchExitMode()` but there's no guarantee `initHighScoreSonas()` was called because that's in `cosCrunchMenu()`. It's not very symmetrical, and it looks like it could be initialized multiple times without freeing. I think you want to free when you leave the menu, which lends itself well to turning all those `cc->state = X` into a function. Or initialize it when the mode starts?